sklearn/tests/test_pipeline.py

```diff
@@ -518,8 +518,9 @@ def test_feature_union():
 
     # test error if some elements do not support transform
     msg = "All estimators should implement fit and transform.*\\bNoTrans\\b"
+    fs = FeatureUnion([("transform", Transf()), ("no_transform", NoTrans())])
     with pytest.raises(TypeError, match=msg):
-        FeatureUnion([("transform", Transf()), ("no_transform", NoTrans())])
+        fs.fit(X)
```

Comment:
We noticed while working on this PR that the `transform` method is not checked which is the intended check for this test. We tried using `fit_transform()` but it also has a skip case defined which allows it to work without a `transform` method.


Wondering if this should be fixed in a separate issue?

Reply:
The test is checking that during `fit` or `fit_*`, the `__init__` parameters are valid. `NoTrans` does not define a `transform` method so `FeatureUnion` should raise an error.

By convention, `fs.transform` does not validate `__init__` parameters.

> skip case defined which allows it to work without a transform method.

May you clarify what you mean by "skip case"?

Reply:
Thank you for clearing this 👍. On reading your comments and revisiting this, there seems to be no problem. There is no skip case, the `_validate_transformers()` method does go through all the validation as required by the test.